opal/mca/mpool/base/base.h

```diff
@@ -53,7 +53,7 @@ OPAL_DECLSPEC OBJ_CLASS_DECLARATION(mca_mpool_base_selected_module_t);
 OPAL_DECLSPEC mca_mpool_base_component_t* mca_mpool_base_component_lookup(const char* name);
 OPAL_DECLSPEC mca_mpool_base_module_t* mca_mpool_base_module_lookup(const char* name);
 
-OPAL_DECLSPEC mca_mpool_base_module_t *mca_mpool_basic_create (void *base, size_t size, unsigned min_align);
+OPAL_DECLSPEC mca_mpool_base_module_t *mca_mpool_basic_create (mca_mpool_base_module_address_fn_t base, size_t size, unsigned min_align);
```

Comment:
@bosilca @hjelmn Is there a reason this argument was `(void*)` instead of `mca_mpool_base_module_address_fn_t`)?

Reply:
@bosilca @hjelmn ping 😄

Reply:
Let me first state that the current implementation of mca_mpool_basic_create() is incorrect because it uses the base argument both as an ```mpool_base``` (which is a function of type mca_mpool_base_module_address_fn_t) and as ```ptr``` (which is a void*).

I think the correct approach would be leave the argument of ```mca_mpool_basic_create``` as a void, but to change the implementation to store the ```base``` only as ```ptr```, and to have a local static function in ```mpool_base``` (similar to what we do to the other base members in mca_mpool_basic_template [opal/mca/mpool/base/mpool_base_basic.c:76]).

Reply:
We'll snip the mpool "fix" off this PR -- I just created https://github.com/open-mpi/ompi/issues/7221 to track the issue separately.